packages/cardpay-sdk/sdk/scheduled-payment-module.ts

```diff
@@ -669,16 +671,189 @@ export default class ScheduledPaymentModule {
     return [nonce, estimate, payload];
   }
 
-  async schedulePayment(txnHash: string): Promise<SuccessfulTransactionReceipt>;
+  private async schedulePaymentOnChainAndUpdateCrank(
+    hubRootUrl: string,
+    authToken: string,
+    scheduledPaymentId: string,
+    safeAddress: string,
+    moduleAddress: string,
+    gasTokenAddress: string,
+    spHash: string,
+    signature: Signature
+  ) {
+    return new Promise<void>((resolve) => {
+      this.schedulePaymentOnChain(safeAddress, moduleAddress, gasTokenAddress, spHash, signature, {
+        onTxnHash: async (txHash: string) => {
+          await hubRequest(hubRootUrl, `api/scheduled-payments/${scheduledPaymentId}`, authToken, 'PATCH', {
+            data: {
+              attributes: {
+                'creation-transaction-hash': txHash,
+              },
+            },
+          });
+          resolve();
+        },
+      });
+    });
+  }
+
+  async schedulePayment(scheduledPaymentId: string): Promise<{ scheduledPaymentId: string; promise: Promise<void> }>;
   async schedulePayment(
+    safeAddress: string,
+    moduleAddress: string,
+    tokenAddress: string,
+    amount: string,
+    payeeAddress: string,
+    feeFixedUSD: number,
+    feePercentage: number,
+    executionGas: number,
+    maxGasPrice: string,
+    gasTokenAddress: string,
+    salt: string,
+    payAt: number | null,
+    recurringDayOfMonth?: number | null,
+    recurringUntil?: number | null,
+    onScheduledPaymentCreate?: (scheduledPaymentId: string) => unknown
+  ): Promise<{ scheduledPaymentId: string; promise: Promise<void> }>;
```

Comment:
since we pass a callback to get the scheduled payment ID as a parameter, probably we don't also need to return it in the response. when I originally proposed returning the schedule payment ID as a response, i didn't actually realize that we were also providing it to callers via a callback as well. i think we can just pick a single approach for returning this value to the callers (such that they don't have to wait for the txn to be mined).

Reply:
Oh, I forgot to delete this return type! It should be `Promise<void>`. Will change this. I chose to provide the scheduled payment id in a callback so that it's immediately available to the caller once the record gets created in the hub, without waiting for the transaction to be created.

Reply:
Fixed.